kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/CheckpointV2ReadSuite.scala

```diff
@@ -0,0 +1,291 @@
+      s"LOCATION '$path' " +
+      s"TBLPROPERTIES ('delta.checkpointInterval' = '2', 'delta.checkpointPolicy'='v2')")
+    spark.sql(s"INSERT INTO $tbl VALUES (1, 'a'), (2, 'b')")
+    spark.sql(s"INSERT INTO $tbl VALUES (3, 'c'), (4, 'd')")
+    spark.sql(s"INSERT INTO $tbl VALUES (5, 'e'), (6, 'f')")
+  }
+
+  def validateSnapshot(
+      path: String,
+      snapshotFromSpark: Snapshot,
+      strictFileValidation: Boolean = true,
+      ckptVersionExpected: Option[Int] = None,
+      expectV2CheckpointFormat: Boolean = true): Unit = {
+    // Create a snapshot from Spark connector and from kernel.
+    val snapshot = latestSnapshot(path)
+    val snapshotImpl = snapshot.asInstanceOf[SnapshotImpl]
+
+    // Validate metadata/protocol loaded correctly from top-level v2 checkpoint file.
+    val expectedMetadataId =
+      DeltaTable.forPath(path).detail().select("id").collect().head.getString(0)
+    assert(snapshotImpl.getMetadata.getId == expectedMetadataId)
+    assert(snapshotImpl.getProtocol.getMinReaderVersion ==
+      snapshotFromSpark.protocol.minReaderVersion)
+    assert(snapshotImpl.getProtocol.getMinWriterVersion ==
+      snapshotFromSpark.protocol.minWriterVersion)
+    assert(snapshotImpl.getProtocol.getReaderFeatures.asScala.toSet ==
+      snapshotFromSpark.protocol.readerFeatureNames)
+    assert(snapshotImpl.getProtocol.getWriterFeatures.asScala.toSet ==
+      snapshotFromSpark.protocol.writerFeatureNames)
+    assert(snapshot.getVersion(defaultTableClient) == snapshotFromSpark.version)
+
+    // Validate that snapshot read from most recent checkpoint.
+    val expectedV2CkptToRead =
+      ckptVersionExpected.getOrElse(snapshotFromSpark.version - (snapshotFromSpark.version % 2))
+    assert(snapshotImpl.getLogSegment.checkpoints.asScala.map(
+      f => FileNames.checkpointVersion(new Path(f.getPath)))
+      .contains(expectedV2CkptToRead))
+    // UUID checkpoints will contain "-" in the name.
+    assert(snapshotImpl.getLogSegment.checkpoints.asScala.map(
+      f => new Path(f.getPath).getName.contains("-")).contains(expectV2CheckpointFormat))
+
+
+    // Validate AddFiles from sidecars found against Spark connector.
+    val scan = snapshot.getScanBuilder(defaultTableClient).build()
+    val foundFiles =
+      collectScanFileRows(scan).map(InternalScanFileUtils.getAddFileStatus).map(
+        _.getPath.split('/').last).toSet
+    val expectedFiles = snapshotFromSpark.allFiles.collect().map(_.path).toSet
+    if (strictFileValidation) {
+      assert(foundFiles == expectedFiles)
+    } else {
+      assert(foundFiles.subsetOf(expectedFiles))
+    }
+  }
+
+  test("v2 checkpoint support") {
+    supportedFileFormats.foreach { format =>
+      withTempDir { path =>
+        val tbl = "tbl"
+        withTable(tbl) {
+          // Create table.
+          withSQLConf(
+            "delta.kernel.default.parquet.reader.batch-size" -> "10",
+            "delta.kernel.default.json.reader.batch-size" -> "10",
```

Comment:
Setting configs like this works for Kernel?

I thought we need to create a custom table client with `Configuration` containing these. And pass that custom table client to `checkTable`.

Reply:
Fixed.